examples/workflows/workspace_workflow.py

```diff
@@ -0,0 +1,35 @@
+from hera.workflows.container_set import ContainerNode, ContainerSet
+from hera.workflows.models import Artifact, ValueFrom, VolumeMount
+from hera.workflows.parameter import Parameter
+from hera.workflows.volume import EmptyDirVolume
+from hera.workflows.workflow import Workflow
+
+with Workflow(
+    generate_name="workspace-",
+    labels={"workflows.argoproj.io/test": "true"},
+    annotations={
+        "workflows.argoproj.io/description": "This workflow demonstrates using a workspace to share files between containers. "
+        'This also allows containers not called "main" to create output artifacts.',
+        "workflows.argoproj.io/version": ">= 3.1.0",
+    },
+) as w:
+    ContainerSet(
+        name="main",
+        containers=[
+            ContainerNode(
```

Comment:
Could we have container node be added to the set similar to how we add tasks to Dag or steps to Steps? (with the context manager)